### DISCUSSION TITLE
Simplify creation of unnamed group when using a custom `NavigationBuilder`

### DISCUSSION THREAD

**Discussion by jacksleight** [filamentphp/filament#1785] [General]
I'm just posting for discussion really, I'm not sure what the solution is, or if one is _really_ needed.

By default Filament creates an unnamed group where it places all of the ungrouped resource links. And this group appears at the top, before any named groups.

But in order to replicate the same with a custom `NavigationBuilder` it’s not entirely straightforward or intuitive.

If you do this:

```php
return $builder
    ->group('', [
        ...DevelopmentResource::getNavigationItems(),
        ...PropertyResource::getNavigationItems(),
    ])
    ->group('Administration', [
        ...StaffResource::getNavigationItems(),
    ]);
```

The unnamed group gets replaced with an empty array of items in `FilamentManager::buildNavigation()`

If you do this:

```php
return $builder
    ->items([
        ...DevelopmentResource::getNavigationItems(),
        ...PropertyResource::getNavigationItems(),
    ])
    ->group('Administration', [
        ...StaffResource::getNavigationItems(),
    ]);
```

The unnamed group appears last, after the named groups.

The only way I have managed to get the desired result is to do this, which feels a bit clunky:

```php
return $builder
    ->group('')
    ->items([
        ...DevelopmentResource::getNavigationItems(),
        ...PropertyResource::getNavigationItems(),
    ])
    ->group(‘Administration’, [
        ...StaffResource::getNavigationItems(),
    ]);
```

But perhaps that's the best way... If so maybe it should be in the docs?

**Comment by danharrin** [comment]
Moved to #1786.
